cmd/updater/main.go

```diff
@@ -934,15 +951,46 @@ func updateOnce(ctx context.Context, opt options) {
 		}
 		groups <- *tg
 	} else { // All groups
-		for _, tg := range cfg.TestGroups {
+		start := time.Now()
```

Comment:
There's a few long functions in this file (though the comments help a lot in tracking what they're doing), consider breaking them up?